src/VisualStudio/Core/Def/Implementation/EditAndContinue/VsENCRebuildableProjectImpl.cs

```diff
@@ -291,6 +292,17 @@ public int StartDebuggingPE()
             }
         }
 
+        private bool AllowsReadOnly(DocumentId documentId)
+        {
+            // All documents of regular running projects are read-only until the debugger breaks the app.
+            // However, ASP.NET doesn�t want Razor views to be read-only, so they can be editable while
+            // the code is running and get refreshed next time the web page is hit.
+            
+            // Note that Razor projects are modelled as a ContainedDocument instead of a StandradTextDocument.
+            var containedDocument = _vsProject.VisualStudioWorkspace.GetHostDocument(documentId) as ContainedDocument;
```

Comment:
These comments all talk just about Razor, but this check will also cover .aspx files.  Please be explicit about whether you want *all* contained languages, or just razor (.cshtml/.vbhtml) files.